api/v1/api.go

```diff
@@ -68,9 +88,12 @@ func (w Wrapper) StoreSecret(ctx context.Context, request StoreSecretRequestObje
 			Title:   "Could not retrieve stored secret",
 		}), nil
 	}
-	return StoreSecret200JSONResponse{Data: Secret(result)}, nil
+	return StoreSecret200JSONResponse{Secret: Secret(result)}, nil
 }
 
 func (w Wrapper) HealthCheck(ctx context.Context, request HealthCheckRequestObject) (HealthCheckResponseObject, error) {
+	if err := w.vault.Ping(); err != nil {
+		return HealthCheck503JSONResponse{Status: Fail}, nil
```

Comment:
Do you want to provide additional information on the fail? eg: invalid creds or timeout or host not found?